extensions/common/auth/auth-delegated/src/main/java/org/eclipse/edc/api/auth/delegated/DelegatedAuthenticationService.java

```diff
@@ -0,0 +1,96 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.api.auth.delegated;
+
+import org.eclipse.edc.api.auth.spi.AuthenticationService;
+import org.eclipse.edc.keys.spi.PublicKeyResolver;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.token.spi.TokenValidationRulesRegistry;
+import org.eclipse.edc.token.spi.TokenValidationService;
+import org.eclipse.edc.web.spi.exception.AuthenticationFailedException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static jakarta.ws.rs.core.HttpHeaders.AUTHORIZATION;
+
+public class DelegatedAuthenticationService implements AuthenticationService {
+
+    public static final String MANAGEMENT_API_CONTEXT = "management-api";
+    @Deprecated
+    private static final String X_API_KEY = "x-api-key";
+    public static final String OLD_API_KEY_WARNING = ("Header '%s' found with the DelegatedAuthenticationService. " +
+            "Please migrate to using the '%s' header at your earliest convenience, this compatibility feature will be removed in upcoming releases!").formatted(X_API_KEY, AUTHORIZATION);
+    private final PublicKeyResolver publicKeyResolver;
+    private final Monitor monitor;
+    private final TokenValidationService tokenValidationService;
+    private final TokenValidationRulesRegistry rulesRegistry;
+
+    public DelegatedAuthenticationService(PublicKeyResolver publicKeyResolver,
+                                          Monitor monitor,
+                                          TokenValidationService tokenValidationService,
+                                          TokenValidationRulesRegistry rulesRegistry) {
+        this.publicKeyResolver = publicKeyResolver;
+        this.monitor = monitor;
+        this.tokenValidationService = tokenValidationService;
+        this.rulesRegistry = rulesRegistry;
+    }
+
+    @Override
+    public boolean isAuthenticated(Map<String, List<String>> headers) {
+
+        if (headers == null) {
+            var msg = "Headers were null";
+            monitor.warning(msg);
+            throw new AuthenticationFailedException(msg);
+        }
```

Comment:
can this happen? looking at the callers (only `AuthenticationRequestFilter` I'd say no

Reply:
this could (theoretically) be called from anywhere, not just request filters. We have the same check in the `TokenBased` one, and it's just a bit more defensive IMO.

Reply:
To me it is too defensive because it could happen only because a developer error, not a "flow" error (e.g. a request that brings null headers cannot happen in fact) so an error like this it would be spotted at the CI level.